Source/Core/Core/HW/SI/SI_DeviceGBA.cpp

```diff
@@ -43,42 +43,51 @@ enum EJoybusCmds
   CMD_WRITE = 0x15
 };
 
-constexpr auto BITS_PER_SECOND = 115200;
-constexpr auto BYTES_PER_SECOND = BITS_PER_SECOND / 8;
+constexpr auto GC_BITS_PER_SECOND = 200000;
+constexpr auto GBA_BITS_PER_SECOND = 250000;
+constexpr auto GC_STOP_BIT_NS = 6500;
+constexpr auto GBA_STOP_BIT_NS = 14000;
 constexpr auto SEND_MAX_SIZE = 5, RECV_MAX_SIZE = 5;
 
 // --- GameBoy Advance "Link Cable" ---
 
 static int GetTransferTime(u8 cmd)
 {
-  u64 bytes_transferred = 0;
+  u64 gc_bytes_transferred = 1;
+  u64 gba_bytes_transferred = 1;
+  u64 stop_bits_ns = GC_STOP_BIT_NS + GBA_STOP_BIT_NS;
 
   switch (cmd)
   {
   case CMD_RESET:
   case CMD_STATUS:
   {
-    bytes_transferred = 4;
+    gba_bytes_transferred = 3;
     break;
   }
   case CMD_READ:
   {
-    bytes_transferred = 6;
+    gba_bytes_transferred = 5;
     break;
   }
   case CMD_WRITE:
   {
-    bytes_transferred = 1;
+    gc_bytes_transferred = 5;
     break;
   }
   default:
   {
-    bytes_transferred = 1;
+    gba_bytes_transferred = 0;
```

Comment:
Should gc_bytes_transferred be set to zero here too?

Reply:
No, this is the case where the GBA doesn't reply to whatever the GC sends.

Reply:
Come to think of it though, the GBA stop bit nanoseconds are probably still necessary since that's the timeout for the GC trying to see if there's a reply. I should add those back.